src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,95 @@
 package com.zipcodewilmington.looplabs;
 
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
-}
+
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+        super(intArray);
+    }
+
+    @Override
+    public Integer[] removeDuplicatesExactly(int exactNumberOfDuplications) {
+        StringBuilder sb = new StringBuilder();
+
+        int counter = 0;
```

Comment:
if you declare your counter inside of your for loop, you wont have to reset it each time. On top of that, you don't use the counter outside of the for loop, so this is considered being out of scope.